### PR TITLE
🪟 Sanitize manifest paths for Windows

### PR DESCRIPTION
**Pull request by ethanhs**
Git won't check out paths with invalid file paths, and `:` is illegal on Windows.

### CODE REVIEW

File: manifests/harbor/ocf-harbor-core_Deployment_harbor.yaml [hunk 1]
```diff
@@ -23,8 +23,8 @@ spec:
         checksum/configmap: a0ab4454c2fe1d3f57871e1cf9c8f836a9bc6059aa09c9d2102ee890ebfdec74
         checksum/configmap-envvars: 3b35112645ecb5545620d410ab254ec63da5cd13992f9a4216a629b5af21731d
         checksum/secret: a62624694d8a1c78bb6ba1dbb5e88dfa865bd255a0430148bb98411aaa57f3e0
-        checksum/secret-envvars: b3195cf8e0af4f6a0c3a540169237e2487f9dae6381ca949db7309d86ca4981a
-        checksum/secret-jobservice: 94dfb2c271df0cc15f4aeaad6a25dd05d22764f0c3492216d893c79471d3ff8a
+        checksum/secret-envvars: ffe77be5b81ace74deb2562dcdb6a5aa84f6ffb0d356666c81199253d6fa3d58
```
Comment (nikhiljha) [inline manifests/harbor/ocf-harbor-core_Deployment_harbor.yaml:26]: whyyyyyyyy

File: ocfkube/__init__.py [hunk 2]
```diff
@@ -32,7 +32,10 @@ def write_manifests(objects, appname: str, manifest_dir: Path):
         name = obj["metadata"].get("name", obj["metadata"].get("generateName", None))
         kind = obj["kind"]
         namespace = obj["metadata"].get("namespace", appname)
-        with open(appdir / f"{name}_{kind}_{namespace}.yaml", "w") as f:
+        manifest_name = f"{name}_{kind}_{namespace}.yaml"
+        # sanitize for Windows paths :sob:
+        manifest_name = manifest_name.replace(':', '-')
```
Comment (nikhiljha) [inline ocfkube/__init__.py:37]: Are we sure this substitution is always safe? I know you can have - in kube object names, so maybe we use _ again? This makes our file names not-machine-parseable but it should work...

File: manifests/harbor/ocf-harbor-core_Deployment_harbor.yaml (hunk 1 above)
Comment (ethanhs) [reply manifests/harbor/ocf-harbor-core_Deployment_harbor.yaml:26]: Yeahh it's not great, should I remove these manifest changes from the PR?

**Comment by nikhiljha** [comment]
this code has been rewritten like 20 times so i'm just gonna close this lol

**Comment by nikhiljha** [comment]
but thanks

**Pull request closed without merging**